knack/log.py

```diff
@@ -140,22 +130,31 @@ def configure(self, args):
             self._init_logfile_handlers(root_logger, cli_logger)
             get_logger(__name__).debug("File logging enabled - writing logs to '%s'.", self.log_dir)
 
-    def _determine_verbose_level(self, args):
+    def _determine_log_level(self, args):
         """ Get verbose level by reading the arguments. """
-        verbose_level = 0
-        for arg in args:
-            if arg == CLILogging.VERBOSE_FLAG:
-                verbose_level += 1
-            elif arg == CLILogging.DEBUG_FLAG:
-                verbose_level += 2
-        # Use max verbose level if too much verbosity specified.
-        return min(verbose_level, len(self.console_log_configs) - 1)
+        # arguments have higher precedence than config
+        if CLILogging.ONLY_SHOW_ERRORS_FLAG in args:
+            if CLILogging.DEBUG_FLAG in args or CLILogging.VERBOSE_FLAG in args:
+                raise CLIError("--only-show-errors can't be used together with --debug or --verbose")
+            self.cli_ctx.only_show_errors = True
+            return 1
+        if CLILogging.DEBUG_FLAG in args:
+            self.cli_ctx.only_show_errors = False
+            return 4
+        if CLILogging.VERBOSE_FLAG in args:
+            self.cli_ctx.only_show_errors = False
+            return 3
+        if self.cli_ctx.only_show_errors:
+            return 1
```

Comment:
is this code possible reached, I think it already return 1 in ( if CLILogging.ONLY_SHOW_ERRORS_FLAG in args )

Reply:
This takes effect when no explicit `--verbose`, `--debug` or `--only-show-warnings` is given, but `only_show_errors` is specified in the config. You may refer to `tests/test_log.py` for a enumeration of possible scenarios.